test/EFCore.PG.FunctionalTests/TestModels/Array/ArrayQueryContext.cs

```diff
@@ -29,6 +29,8 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
 
                 e.Property(ae => ae.ValueConvertedList)
                     .HasPostgresArrayConversion(w => -(int)w, v => (SomeEnum)(-v));
+
+                e.HasIndex(ae => ae.NonNullableIndexedText);
```

Comment:
Unless I'm mistaken, you can simply define the index over e.g. NullableText and repro this with NullableText and NonNullableText - this way we don't have to add a new property and update all the SQL assertions.

Reply:
Rewrited as you suggested